app/models/random_access_key.rb

```diff
@@ -0,0 +1,3 @@
+class RandomAccessKey < ApplicationRecord
+  belongs_to :request
```

Comment:
Based on my suggest you would create the `access_key` when this method is initialized.  Suggest using a before callback.